src/circuit.rs

```diff
@@ -249,26 +314,30 @@ mod tests {
         let mut circuit = TestCircuit::default();
 
         // Compile the circuit
-        let (pk_p, vk_p, pi_pos) = circuit.compile(&pp)?;
+        let (pk_p, og_verifier_data) = circuit.compile(&pp)?;
 
         // Write the keys
         File::create(&pk_path)
             .and_then(|mut f| f.write(pk_p.to_var_bytes().as_slice()))
             .unwrap();
-        File::create(&vk_path)
-            .and_then(|mut f| f.write(&vk_p.to_bytes()))
-            .unwrap();
 
         // Read ProverKey
         let pk = fs::read(pk_path).unwrap();
         let pk = ProverKey::from_slice(pk.as_slice())?;
 
-        // Read VerifierKey
-        let vk = fs::read(vk_path).unwrap();
-        let vk = VerifierKey::from_slice(vk.as_slice())?;
-
         assert_eq!(pk, pk_p);
-        assert_eq!(vk, vk_p);
+
+        // Store the VerifierData just for the verifier side:
+        // (You could also store pi_pos and VerifierKey sepparatedly).
+        File::create(&vd_path)
+            .and_then(|mut f| {
+                f.write(og_verifier_data.to_var_bytes().as_slice())
+            })
+            .unwrap();
```

Comment:
Avoids `unwrap` whenever is possible even in tests, just use an `expect` or makes the test returns a `Result`.